src/components/MailboxPicker.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+	<Treeselect
+		ref="Treeselect"
+		:options="mailboxes"
+		:multiple="false"
+		:clearable="false"
+		v-bind="$attrs"
+		v-on="$listeners" />
+</template>
+<script>
+import Treeselect from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+
+export default {
+	name: 'MailboxPicker',
+	components: {
+		Treeselect,
+	},
+	props: {
+		accountid: {
+			type: Number,
+			required: true,
+		},
+	},
+	computed: {
+		mailboxes() {
+			const boxes = []
+			this.getMailboxes(boxes)
+			return boxes
+		},
+	},
+	methods: {
+		getMailboxes(root, folderid) {
+			let folders = []
+			if (!folderid) {
+				folders = this.$store.getters.getMailboxes(this.accountid)
+			} else {
+				folders = this.$store.getters.getSubMailboxes(folderid)
+			}
+			folders.forEach((folder) => {
+				const mailbox = {
+					id: folder.databaseId,
+					label: folder.displayName,
+				}
+				if (folder.mailboxes.length > 0) {
+					mailbox.children = []
+					this.getMailboxes(mailbox.children, mailbox.id)
+				}
+				root.push(mailbox)
```

Comment:
I would push the parent first, then append the children. IMO it should be

* Archive
* Archive.2020

and not

* Achive.2020
* Achive

Reply:
this is obsolete